sale_last_price_info/models/product.py

```diff
@@ -0,0 +1,29 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.one
+    def _get_last_sale(self):
+        """ Get last sale price, last sale date and last customer """
+        lines = self.env['sale.order.line'].search(
+            [('product_id', '=', self.id),
+             ('state', 'in', ['confirmed', 'done'])]).sorted(
+            key=lambda l: l.order_id.date_order, reverse=True)
+        self.last_sale_date = lines[:1].order_id.date_order
+        self.last_sale_price = lines[:1].price_unit
+        self.last_customer = lines[:1].order_id.partner_id
+
+    last_sale_price = fields.Float(
+        string='Last Sale Price', compute='_get_last_sale')
+    last_sale_date = fields.Date(
+        string='Last Sale Date', compute='_get_last_sale')
+    last_customer = fields.Many2one(
```

Comment:
2015-06-26 0:49 GMT-05:00 Pedro M. Baeza <notifications@github.com>:

> No, we use this naming convention because we prefer it over the _id one.
> Code without these suffixes is clearer, and there's no too much advantage
> in using them (the type can be easily inspected in lot of places).
>

I think _id have more sense for many2one name of fields (even if yes it is
an object now), introduce new "conventions" becomes thing difficult to
inspect with pure greps +1 for the _id.


-- 
--------------------
Saludos Cordiales

CEO at Vauxoo <https://www.vauxoo.com> Odoo's Gold Partner.

[image: --]
Nhomar Hernandez
[image: http://]about.me/nhomar
<http://about.me/nhomar?promo=email_sig>